src/rebase.c

```diff
@@ -987,6 +1019,11 @@ static int rebase_commit__create(
 	if (error < 0)
 		git_commit_free(commit);
 
+	if (signature)
+		free(signature);
+	if (signature_field)
+		free(signature_field);
```

Comment:
We can't free memory that the user allocated because we don't know how it was allocated.  Win32, for example, has several different heap allocators.  Concrete example: if libgit2 is compiled as a release library and the application calling it is compiled as a debug binary, then the `malloc` from the application will be incompatible with the `free` in the library.

Instead, pass a `git_buf` to the callback and let them [`git_buf_set`](https://github.com/libgit2/libgit2/blob/master/include/git2/buffer.h#L117) it.